src/models/base.py

```diff
@@ -174,6 +194,34 @@ def evaluate(self, save_results: bool = True, save_preds: bool = False) -> None:
                     .to_xarray()
                 )
 
+                if self.predict_delta:
+                    # get the NON-NORMALIZED data (ModelArrays)
+                    dl = self.get_dataloader(
```

Comment:
Is is necessary to call a new dataloader here?

An alternative is to add an attribute to the `ModelArrays`, which (if `predict_delta == True`) contains the appropriate target xarray ( / numpy array - that might be easier).

That way, all of this can be abstracted away into the dataloader, and it should be a little cleaner since all the data will be there together.

Reply:
where would i pick up the data from the ModelArrays though? because aren't the `TestData` objects only opened in the `predict()` functions e.g. [here](https://github.com/esowc/ml_drought/blob/95b03918bbcfa751d32ee401b87c6535aae2b396/src/models/neural_networks/base.py#L168)

Reply:
right - this would be in the dataloader step. That test arrays loader returns ModelArrays (https://github.com/esowc/ml_drought/blob/95b03918bbcfa751d32ee401b87c6535aae2b396/src/models/data.py#L662).

Reply:
Nice thanks for your help here Gabi! Much more sensible now I think (plus we're passing around numpy arrays instead of xarray objects)